### DISCUSSION TITLE
D1 Mini - powering a USB peripheral via the 5 volt onboard pin

### DISCUSSION THREAD

**Discussion by EnnPeeTee** [micropython/micropython#11991] [ESP8266]
I am using a D1 Mini [ ESP-8266-12F micro-USB port ] with a USB GPS ‘mouse’ plugged into the on-board USB port.  I have software loaded to send the GPS serial data via Wi-Fi to an MQTT Broker for use with Node-Red.

5 volt power is applied to the D1 via the 5v and GND pins and I was hoping that the 5 volts would feed back to the USB connector to power the GPS unit which takes 85mA at 5 volts.  However, it didn’t like it!

Is the 5 volt pin on the main board connected directly to the 5v pin on the USB connector ?

Any guidance will be greatly appreciated.

**Comment by scruss** [comment]
That would be a question for the hardware designer. There are many versions of the D1 Mini, from the originals by [wemos](https://www.wemos.cc/en/latest/d1/d1_mini_lite.html) down to a near infinite number of much lower quality clones.

I'm surprised that you're able to get GPS data via the onboard micro-USB port. It's not a USB host, and the USB-to-UART chip on the D1 mini can't pretend to be a USB host. The D1 Mini Lite schematic shows a diode between the USB power line and the +5V terminal to explicitly prevent the kind of backfeed you want.

**Reply by EnnPeeTee** [reply]
Aha!  Thank you for that!   I shall have to look at breaking out the USB lines and feeding the data lines to the appropriate pins and manage them through the s/w.  

Whilst the s/w flags as syntax OK, it hasn't been tested live yet because I had to unplug the USB lead to the Arduino IDE to allow me to plug in the GPS.  I think this one is dead now.  

Back to the drawing board!

Many thanks again.

**Reply by scruss** [reply]
what I said was, you won't be able to read a USB port via the port on the D1 mini.

**Reply by EnnPeeTee** [reply]
I should have been more specific - too late at night etc.

I shall break out the USB lines from the GPS and assign the 2 data lines to appropriate ports on the D1 and power both conventionally. The D1's on-board USB can be used purely for programming.

**Comment by bidrohini** [comment]
The D1 Mini board has a built-in 5V to 3.3V voltage regulator that supplies power to the ESP8266 module and other components on the board. The VIN pin is the input to this voltage regulator, and it can accept an input voltage range of about 4.5V to 9V. The 5V pin on the board is the regulated output from the voltage regulator, providing a stable 5V output.

If you want to power external devices, like your USB GPS 'mouse', you should not try to draw power directly from the USB connector. Instead, you should connect the 5V output from the D1 Mini (the 5V pin on the board) to the power input of your GPS device. Make sure to check the power requirements of your GPS device to ensure it is within the acceptable range of the D1 Mini's voltage regulator.

Here's how you can connect the GPS device to the D1 Mini:

Connect the 5V pin (output) on the D1 Mini board to the 5V power input of your USB GPS 'mouse'.
Connect the GND pin on the D1 Mini board to the GND of your USB GPS 'mouse'.

Here is a D1 mini based PCB. It uses SN74LVC2T45DCUR level shifter. The board supports 5-12V input. It has jumper wires to match the input voltage. 

https://www.pcbway.com/project/shareproject/WeMos_D1_mini_WLED_Controller_5_12V_2d4e882a.html

**Reply by EnnPeeTee** [reply]
Thank you for that very useful info,  

The only info I have on the GPS power consumption is "45mA during data acquisition" at 5v. I don't think it is safe to assume that is the max current for there may be a large inrush at start up so I shall play it safe and power both the GPS and the D1 from the same 5v source directly.

I shall have to pay attention to the two data lines from the GPS when connecting to the 3v3 IO pins on the D1 - probably with a couple of potential dividers using resistors.
